src/shared/db/wallets/get-wallet-details.ts

```diff
@@ -92,13 +92,33 @@ const smartGcpKmsWalletSchema = gcpKmsWalletSchema
   .merge(smartWalletPartialSchema)
   .merge(baseWalletPartialSchema);
 
+const circleWalletSchema = z
+  .object({
+    type: z.literal("circle"),
+    platformIdentifiers: z.object({
+      circleWalletId: z.string(),
+      walletSetId: z.string(),
+    }),
+    credentialId: z.string(),
+  })
+  .merge(baseWalletPartialSchema);
+
+const smartCircleWalletSchema = circleWalletSchema
+  .extend({
+    type: z.literal("smart:circle"),
+  })
+  .merge(smartWalletPartialSchema)
+  .merge(baseWalletPartialSchema);
```

Comment:
are these merges necessary if we're already extending circleWalletSchema which contains these already?